x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/CsvTests.java

```diff
@@ -470,6 +471,13 @@ private void assertWarnings(List<String> warnings) {
                 normalized.add(normW);
             }
         }
-        assertMap(normalized, matchesList(testCase.expectedWarnings(true)));
+        List<Pattern> expectedWarningsRegex = testCase.expectedWarningsRegex();
```

Comment:
Don't know if deps allow it, would be nice if we could reuse the same code from `RestEsqlTestCase`, or the other way around.

Reply:
The two classes don't see each other, but I can move the logic to `EsqlTestUtils`

Reply:
> The two classes don't see each other

Ah, ok, then up to you, it's a small duplication, can stay as is too.